src/Rules/IRuleInfo.cs

```diff
@@ -0,0 +1,104 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+// Note: we have multiple enum definitions for IssueSeverity and IssueType in SLVS
+// They should be consolidated if possible.
+// See https://github.com/SonarSource/sonarlint-visualstudio/issues/3617
```

Comment:
I defined completely new types for severity and issue type, and a new `IRuleInfo` class that is a superset of `IRuleHelp`.

In a later PR, I'll replace `IRuleHelp` with `IRuleInfo`. I didn't want to do that in this PR as it would create too much churn.